driver/manager.go

```diff
@@ -0,0 +1,70 @@
+package driver
+
+import (
+	"context"
+	"sort"
+
+	dockerclient "github.com/docker/docker/client"
+	"github.com/pkg/errors"
+)
+
+type Factory interface {
+	Name() string
+	Usage() string
+	Priority() int // take initConfig?
+	New(ctx context.Context, cfg InitConfig) (Driver, error)
+}
+
+type BuildkitConfig struct {
+	// Entitlements []string
+	// Rootless bool
+}
+
+type InitConfig struct {
+	// This object needs updates to be generic for different drivers
+	Name           string
+	DockerAPI      dockerclient.APIClient
+	BuildkitConfig BuildkitConfig
+	Meta           map[string]interface{}
+}
+
+var drivers map[string]Factory
```

Comment:
`var drivers = map[string]Factory{}`

Reply:
there is a nil check on line 34

Reply:
Hence my suggestion, so you can get rid of that :)